qiskit/providers/ibmq/job/ibmqjob.py

```diff
@@ -344,8 +344,15 @@ def error_message(self):
 
         if not self._api_error_msg:
             job_response = self._get_job()
-            results = job_response['qObjectResult']['results']
-            self._api_error_msg = build_error_report(results)
+            if 'qObjectResult' in job_response:
+                results = job_response['qObjectResult']['results']
+                self._api_error_msg = build_error_report(results)
+            elif 'qasms' in job_response:
+                qasm_statuses = [qasm['status'] for qasm in job_response['qasms']]
+                self._api_error_msg = 'Job resulted in the following QASM status(es): ' \
+                                      '%s.' % ', '.join(qasm_statuses)
```

Comment:
Could you use [`.format()`](https://docs.python.org/3/library/string.html#string.Formatter.format) instead of `%` string formatting?